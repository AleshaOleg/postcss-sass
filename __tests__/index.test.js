var postcss = require('postcss');
var gonzales = require('gonzales-pe');

var getPostCssTreeFromSass = require('../helpers/getPostCssTreeFromSass');
var getCss = require('../helpers/getCss');

it('PostCSS dependency', function () {
    expect(typeof postcss).toBe('function');
});

it('gonzales-pe dependency', function () {
    expect(typeof gonzales).toBe('object');
});

console.log(getPostCssTreeFromSass('alt').nodes[0]);
console.log(getCss('alt').nodes[0]);

