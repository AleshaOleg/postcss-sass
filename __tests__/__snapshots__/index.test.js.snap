exports[`test basic.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "nodes": Array [
        Object {
          "prop": "font",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 13,
              "line": 4,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 3,
              "line": 4,
            },
          },
          "type": "decl",
          "value": "Arial",
        },
        Object {
          "prop": "background",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 18,
              "line": 5,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 3,
              "line": 5,
            },
          },
          "type": "decl",
          "value": "blue",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "body",
      "source": Object {
        "end": Object {
          "column": 19,
          "line": 5,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 1,
          "line": 3,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "width",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 14,
              "line": 8,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 3,
              "line": 8,
            },
          },
          "type": "decl",
          "value": "700px",
        },
        Object {
          "prop": "height",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 13,
              "line": 9,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 3,
              "line": 9,
            },
          },
          "type": "decl",
          "value": "100",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "#page",
      "source": Object {
        "end": Object {
          "column": 1,
          "line": 15,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 1,
          "line": 7,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "height",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 17,
              "line": 11,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 5,
              "line": 11,
            },
          },
          "type": "decl",
          "value": "300px",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "#page #header",
      "source": Object {
        "end": Object {
          "column": 1,
          "line": 15,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 3,
          "line": 10,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "font-size",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 21,
              "line": 13,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 7,
              "line": 13,
            },
          },
          "type": "decl",
          "value": "50px",
        },
        Object {
          "prop": "color",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 21,
              "line": 14,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 7,
              "line": 14,
            },
          },
          "type": "decl",
          "value": "blue",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "#page #header h1",
      "source": Object {
        "end": Object {
          "column": 22,
          "line": 14,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 5,
          "line": 12,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "width",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 18,
              "line": 19,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 7,
              "line": 19,
            },
          },
          "type": "decl",
          "value": "100px",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "#content.user.show #container.top #column.left",
      "source": Object {
        "end": Object {
          "column": 19,
          "line": 19,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 5,
          "line": 18,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "width",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 18,
              "line": 21,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 7,
              "line": 21,
            },
          },
          "type": "decl",
          "value": "600px",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "#content.user.show #container.top #column.right",
      "source": Object {
        "end": Object {
          "column": 19,
          "line": 21,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 5,
          "line": 20,
        },
      },
      "type": "rule",
    },
    Object {
      "nodes": Array [
        Object {
          "prop": "background",
          "raws": Object {
            "before": "",
            "between": ": ",
          },
          "source": Object {
            "end": Object {
              "column": 21,
              "line": 23,
            },
            "source": Input {
              "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
              "id": "<input css 3>",
            },
            "start": Object {
              "column": 5,
              "line": 23,
            },
          },
          "type": "decl",
          "value": "brown",
        },
      ],
      "raws": Object {
        "between": "
",
      },
      "selector": "#content.user.show #container.bottom",
      "source": Object {
        "end": Object {
          "column": 22,
          "line": 23,
        },
        "source": Input {
          "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
          "id": "<input css 3>",
        },
        "start": Object {
          "column": 3,
          "line": 22,
        },
      },
      "type": "rule",
    },
  ],
  "raws": Object {
    "after": "",
    "semicolon": false,
  },
  "source": Object {
    "end": Object {
      "column": 22,
      "line": 23,
    },
    "source": Input {
      "css": "

body
  :font Arial
  :background blue

#page
  :width 700px
  :height 100
  #header
    :height 300px
    h1
      :font-size 50px
      :color     blue

#content.user.show
  #container.top
    #column.left
      :width 100px
    #column.right
      :width 600px
  #container.bottom
    :background brown
",
      "id": "<input css 3>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
